src/app/drivingLicence/controllers/root.js

```diff
@@ -1,21 +1,108 @@
 const { Controller: BaseController } = require("hmpo-form-wizard");
 
+const {
+  API: {
+    BASE_URL,
+    PATHS: { PERSON_INFO }
+  }
+} = require("../../../lib/config");
+
 class RootController extends BaseController {
   async saveValues(req, res, next) {
-    const sharedClaims = req.session?.shared_claims;
+    req.sessionModel.reset();
+    req.sessionModel.set("isAuthSourceRoute", false);
 
-    if (sharedClaims) {
-      if (sharedClaims?.names?.length > 0) {
-        req.journeyModel.set("surname", sharedClaims.names[0]?.familyName);
-        req.journeyModel.set("givenNames", sharedClaims.names[0]?.givenNames);
-      }
+    const headers = {
+      session_id: req.session?.tokenId
+    };
+
+    if (process.env.AUTH_SOURCE_ENABLED === "true") {
+      //axios.post to new personInfo endpoint, put licence details into req.session.shared_claims shared claims
+      const personInfoApiResponse = await req.axios.get(
+        `${BASE_URL}/${PERSON_INFO}`,
+        { headers: headers }
+      );
+
+      this.checkForValidSharedClaimsData(req, personInfoApiResponse);
+
+      if (req.sessionModel.get("isAuthSourceRoute") === true) {
+        req.sessionModel.set(
+          "drivingLicenceNumber",
+          personInfoApiResponse.data.drivingPermit[0].personalNumber
+        );
+
+        req.sessionModel.set(
+          "expiryDate",
+          personInfoApiResponse.data.drivingPermit[0].expiryDate
+        );
+
+        req.sessionModel.set(
+          "issueDate",
+          personInfoApiResponse.data.drivingPermit[0].issueDate
+        );
+
+        req.sessionModel.set(
+          "licenceIssuer",
+          personInfoApiResponse.data.drivingPermit[0].issuedBy
+        );
+
+        req.sessionModel.set(
+          "postcode",
+          personInfoApiResponse.data.address[0].postalCode
+        );
+
+        if (personInfoApiResponse.data.drivingPermit[0]?.issuedBy === "DVLA") {
+          req.sessionModel.set(
+            "issueNumber",
+            personInfoApiResponse.data.drivingPermit[0].issueNumber
+          );
+        }
 
-      if (sharedClaims?.dateOfBirths?.length > 0) {
-        req.journeyModel.set("dateOfBirth", sharedClaims.dateOfBirths[0]);
+        this.setNames(req, personInfoApiResponse.data.name[0].nameParts);
+
+        req.sessionModel.set(
+          "dateOfBirth",
+          personInfoApiResponse.data.birthDate[0].value
+        );
       }
     }
     super.saveValues(req, res, next);
   }
+
+  async checkForValidSharedClaimsData(req, personInfoApiResponse) {
+    if (
+      personInfoApiResponse.data?.drivingPermit[0]?.personalNumber &&
+      personInfoApiResponse.data?.drivingPermit[0]?.expiryDate &&
+      personInfoApiResponse.data?.drivingPermit[0]?.issueDate &&
+      personInfoApiResponse.data?.drivingPermit[0]?.issuedBy &&
+      personInfoApiResponse.data?.drivingPermit[0]?.issueNumber &&
```

Comment:
I think this needs to be more complicated. For instance we only expect an issue number if its DVLA